Formula/rom-tools.rb

```diff
@@ -30,25 +30,61 @@ class RomTools < Formula
   uses_from_macos "expat"
   uses_from_macos "zlib"
 
+  on_linux do
+    depends_on "glm" => :build
+    depends_on "jpeg" => :build
+    depends_on "portaudio" => :build
+    depends_on "portmidi" => :build
+    depends_on "pugixml" => :build
+    depends_on "pulseaudio" => :build
+    depends_on "qt@5" => :build
+    depends_on "rapidjson" => :build
+    depends_on "sdl2_ttf" => :build
+    depends_on "sqlite" => :build
```

Comment:
```ruby
uses_from_macos "sqlite" => :build
```

Reply:
It currently doesn't use system `sqlite` (and some of other dependencies) on macOS unless we set the CMake variable.

I think `rom-tools` essentially does same build as `mame` formula to create another binary, which is only temporary and isn't saved into bottle. So, we would need to add `depends_on "sqlite" => :build` and `USE_SYSTEM_LIB_SQLITE3=1` on macOS to actually enable it.

We could replicate the entire `mame` build, but I am just doing it on Linux for now to see if it helps bring the build time down.

If it actually builds and we want to rewrite `rom-tools` on macOS, then we should probably do it in another PR given that it takes 4+ hours to build both `mame` and `rom-tools` on macOS (and took 9 hours on Linux).

Reply:
I tried to build on my docker machine, I add `depends_on "alsa_lib" => :build` and it build success. I don't have Linux desktop environment so I can't test it in real environment.

In the past, I prefer add `SUBTARGET=dummy` in rom-tools, but it generate different checksum binary, so I decide use same subtarget as MAME, which is spend same time as compiling MAME.

Reply:
`alsa_lib` should allow `rom-tools` to finish building its own copy of `portaudio`.

Most Linux distros have a single package for `mame + mame(rom)-tools` that share the same build, so they will use an identical build environment for both.

The main think I am considering is the fact that we have a single self-hosted Linux, so all builds that take longer or require more disk space than allotted for GitHub ubuntu-latest will bottleneck on it. This can be seen by the `qt` PR that has been stalling for 2+ hours on this PR.

So, I am checking if using our already built system libraries can help reduce build time. If it takes about the same time, then just using bundled dependencies are fine in the case of `rom-tools` as they aren't actually used in final bottle.

The flip side is how dependent testing is done. Homebrew Linux currently doesn't test dependents and (in most cases) the dependents are not built from source, so it shouldn't be that bad in possible added build time.